pkg/cli/codeset/register.go

```diff
@@ -53,6 +56,13 @@ LOCATION can be path to local directory or URL of a git repository`,
 	cmd.Flags().StringVarP(&o.Project, "project", "p", "", "the project to which the codeset belongs")
```

Comment:
Folow-up to my suggestion about using global variables: what would happen if you did this here instead ?

```
cmd.Flags().StringVarP(&gOpt.CurrentProject, "project", "p", "", "the project to which the codeset belongs")
```

that's what I meant by linking the flag to the global option variable. This way, you can configure this on a per-command basis instead of aggregating everything in that `cli/cmd.go` common file, transparently from each command. Generally speaking, transparent mechanisms like this are more difficult to debug and maintain.

There's another reason for it: the way that I see it, this CLI code is built on top of something resembling a framework that people can easily use to write new commands. A "current project" is a global CLI feature that can apply to any command, so you should implement something to make it easy for anyone to use it when writing new commands. Something as simple as a function call that says "link this flag to the current project".

Reply:
value of `gOpt.CurrentProject` is not yet known at this time, because it's only filled when parsing the command line (`bindFlags`)

Reply:
ok, let's leave this for some other time, good enough for now